csharp/rocketmq-client-csharp/Producer.cs

```diff
@@ -128,22 +128,18 @@ public async Task<SendReceipt> Send(Message message)
                 {
                     ResourceNamespace = resourceNamespace(),
                     Name = message.Topic
-                }
+                },
+                UserProperties = { message.UserProperties },
+                SystemProperties = new rmq::SystemProperties
+                {
+                    MessageId = message.MessageId,
+                    MessageType = rmq::MessageType.Normal,
+                    Tag = message.Tag ?? string.Empty,
```

Comment:
If message's tag is not set, we should not set it as an empty string.

As you could see from https://github.com/apache/rocketmq-apis/blob/e83a1bceacbef5be465a459965d8edee59cad0c2/apache/rocketmq/v2/definition.proto#L200, the message tag field is optional, which means that it should be absent if message tag is not set.

Reply:
fixed in latest revision